core/device_impl.cpp

```diff
@@ -443,6 +444,86 @@ void DeviceImpl::get_param_float_async(const std::string &name,
                                             callback));
 }
 
+MavlinkCommands::Result DeviceImpl::set_flight_mode(FLIGHT_MODE device_mode)
```

Comment:
I'm not sure if `MavlinkCommands::Result` is the correct result for a flight mode change. I vote that this has its own return type, and then map the `CommandResult` to a `SetFlightMode` result.

Reply:
@julianoes As flight mode change is merely a MAVLink command operation, its result remains same; except for checking unsupported mode. Plugins should only use supported modes; but we are handling anyway for unsupported mode before sending command further. I think it should be fine. What do you think ?

Reply:
I would vote to just return `MavlinkCommand::ERROR` for the case of an unsupported mode, and add a `LogErr() << "Unsupported mode".

The reason is that any plugin using any kind of command, now needs to catch the command error `UNSUPPORTED_MODE` even for commands that have nothing to do with mode, and that is not intuitive.